modules/services/MapRouletteService.js

```diff
@@ -0,0 +1,384 @@
+                .then(response => response.json())
+                .then(challengeData => {
+                  // Update the cache with the deleted status of the challenge
+                  this._deletedChallenges[challengeData.id] = challengeData.deleted;
+                  // if the challenge is deleted, skip the rest of the loop iteration
+                  if (challengeData.deleted) return;
+
+                  const id = task.id;
+                  let loc = [task.point.lng, task.point.lat];
+                  loc = this._preventCoincident(loc);
+                  const itemType = task.type;
+
+                  let d = new MapRouletteTask(loc, this, id, { task, type: itemType });
+                  this._cache.tasks.set(d.id, d);
+                  this._cache.rtree.insert(this._encodeIssueRtree(d));
+                });
+            }
+          }
+
+          this.context.deferredRedraw();
+          this.emit('loadedData');
+        })
+        .catch(err => {
+          if (err.name === 'AbortError') return;    // ok
+          this._cache.loadedTile[tile.id] = true;   // don't retry
+        })
+        .finally(() => {
+          delete this._cache.inflightTile[tile.id];
+        });
+    }
+  }
+
+
+  /**
+   * loadTaskDetailAsync
+   * @param   task
+   * @return  Promise
+   */
+  loadTaskDetailAsync(task) {
+    if (task.details !== undefined) return Promise.resolve(task);
+    const url = `${MAPROULETTE_API}/challenge/${task.task.parentId}`;
+    const handleResponse = (data) => {
+      task.instruction = marked.parse(data.instruction) || '';
+      task.details = marked.parse(data.description) || '';
+      this._deletedChallenges[data.id] = data.deleted;
+    };
+
+    return fetch(url)
+      .then(utilFetchResponse)
+      .then(handleResponse)
+      .then(() => task);
+  }
+
+
+  /**
+   * postUpdate
+   * @param   task
+   * @param   callback
+   */
+  postUpdate(task, callback) {
+    const context = this.context;
+    const sidebar = context.systems.ui.sidebar;
+    if (this._cache.inflightPost[task.id]) {
+      console.log('Task update already inflight for task:', task);
+      return callback({ message: 'Issue update already inflight', status: -2 }, task);
+    }
+    const commentUrl = `${MAPROULETTE_API}/task/${task.id}/comment`;
+    const updateTaskUrl = `${MAPROULETTE_API}/task/${task.id}/${task.taskStatus}`;
+    const releaseTaskUrl = `${MAPROULETTE_API}/task/${task.taskId}/release`;
+    const controller = new AbortController();
+    this._cache.inflightPost[task.id] = controller;
+
+    fetch(commentUrl, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        'apiKey': task.mapRouletteApiKey
+      },
+      body: JSON.stringify({ actionId: 2, comment: task.comment }),
+      signal: controller.signal
+    })
+    .then(response => {
+      if (!response.ok) throw new Error(`Error posting comment: ${response.statusText}`);
+      return response.json();
+    });
```

Comment:
If i'm reading it correctly, this fetch looks abortable, but there isn't a `catch` that handles that situation?   It could also be that we never abort it though elsewhere in the code so it doesn't need the AbortController?

```js
      .catch(err => {
        if (err.name === 'AbortError') return;          // ok
        if (err instanceof Error) console.error(err);   // eslint-disable-line no-console
        // handle the failure of the post?
      })
      .finally(() => {
        delete this._cache.inflightPost[task.id];    // it's not inflight anymore
      });
```